src/v/cloud_roles/refresh_credentials.cc

```diff
@@ -91,43 +89,62 @@ load_and_validate_env_var(std::string_view env_var) {
     return std::nullopt;
 }
 
+static std::optional<net::unresolved_address>
+parse_address(std::string_view maybe_address) {
+    auto separator = maybe_address.find(':');
+    if (
+      separator == std::string_view::npos || separator == 0
+      || separator == maybe_address.size() - 1) {
+        vlog(
+          clrl_log.warn,
```

Comment:
nit: make this an error so if there's a typo in a test, it'll be more obvious why a node doesn't have the right credentials

Reply:
fixed to throw exception